kie-wb-common-react/boxed-expression-editor/src/components/ExpressionContainer/ExpressionContainer.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from "react";
+import {useState} from "react";
+import * as _ from "lodash";
+import "./ExpressionContainer.css"
+import {useBoxedExpressionEditorI18n} from "../../i18n";
+import {
+  Dropdown,
+  DropdownItem,
+  KebabToggle,
+  Popover,
+  SimpleList,
+  SimpleListItem,
+  SimpleListItemProps
+} from "@patternfly/react-core";
+
+export interface ExpressionContainerProps {
+  /** The name of the expression */
+  name: string,
+  /** The type of the expression */
+  type?: string,
+  /** Selected expression is already present */
+  selectedExpression?: string
+}
+
+const ExpressionContainer: (props: ExpressionContainerProps) => JSX.Element = (props: ExpressionContainerProps) => {
+  const {i18n} = useBoxedExpressionEditorI18n();
+
+  const [logicTypeIsPresent, isLogicTypeSet] = useState(!_.isEmpty(props.selectedExpression));
+  const [actionDropdownIsOpen, isActionDropdownOpen] = useState(false);
+
+  const hideSelectorMenuPopover = () => {
+    const elem = document.querySelector('.selector-menu .pf-c-button');
+    const button: HTMLButtonElement = elem as HTMLButtonElement;
+    return button.click();
+  };
+
+  const onLogicTypeSelect = (currentItem: React.RefObject<HTMLButtonElement>, currentItemProps: SimpleListItemProps) => {
+    isLogicTypeSet(true);
+    document.getElementById("expression-container-box")!.innerHTML = currentItemProps.children as string;
```

Comment:
Instead of manipulating DOM elements directly, could we handle "sub-React-components" for each boxed expression type (that would be more React-friendly too)? :-)